### DISCUSSION TITLE
Inertia in k-means clustering

### DISCUSSION THREAD

**Discussion by frederik-f** [GAA-UAM/scikit-fda#594] [Q&A] [answered]
Dear all, 

I'm using the k-means algorithm for clustering a set of curves. Does anybody know, how the inertia is exactly calculated in this case? Suppose I have the grid_points as [0, 1, 2, 3] and the data_matrix as [[1, 2, 3, 4], [6, 5, 4, 3], [5, 3, 1, -1]]. If I'm using 2 clusters, I would e.g. retrieve the cluster_centers = [[5.5, 1],[4, 2],[2.5, 3],[1, 4]] and an inertia of 10.5, which is the sum of the squared distances. The single distances read [0., 2.29128785, 2.29128785]. I would really appreciate, if anybody could tell me how these distances are calculated exactly, ideally for this specific example. 

Thanks a lot in advance!

**Comment by vnmabus** [comment, marked as answer]
In your case you have the discretization grid $\mathbf{t} = (0, 1, 2, 3)$ and the functions $x_0, x_1, x_2$ with $x_0(\mathbf{t}) = (1, 2, 3, 4)$, $x_1(\mathbf{t}) = (6, 5, 4, 3)$ and $x_2(\mathbf{t}) = (5, 3, 1, -1)$.
The cluster centers are $c_0, c_1$ with $c_0(\mathbf{t}) = (5.5, 4, 2.5, 1)$ and $c_1(\mathbf{t}) = (1, 2, 3, 4)$.
The distance between two functions $f$ and $g$ is configurable using the `metric` parameter. By default is the $L^2$ distance, that is $d(f, g) = \sqrt{\int_{\mathcal{T}} |f(t)-g(t)|^2dt}$ (thus, analog to the Euclidean distance).
The distances appearing in the inertia are those between each observation and the cluster it belongs to (the closest one). Thus:
$d_0 = d(x_0, c_1) = 0$ (because both are equal)
$d_1 = d(x_1, c_0) \sim 2.29128785$
$d_2 = d(x_2, c_0) \sim 2.29128785$
In all cases the integral is approximated using a Simpson quadrature rule, as implemented in [`scipy.integrate.simpson`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.simpson.html).

**Reply by frederik-f** [reply]
Thanks a lot for your quick response and detailed explanation, which fully resolved my question!
